### PR TITLE
metastore transaction solution 3

### PR DESCRIPTION
**Pull request by wfxr**

### CODE REVIEW

File: internal/database/metadata/txproxy.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	UpdateFeatureGroupTx func(context.Context, *sqlx.Tx, UpdateFeatureGroupOpt) error
+
+	CreateRevisionTx func(context.Context, *sqlx.Tx, CreateRevisionOpt) (int32, string, error)
+	UpdateRevisionTx func(context.Context, *sqlx.Tx, UpdateRevisionOpt) error
+}
+
+func (tp *TxProxy) WithTransaction(ctx context.Context, fn func(tx *Tx) error) error {
+	tx, err := tp.BeginTxFn(ctx, nil)
+	if err != nil {
+		return nil
```
Comment (YikSanChan) [inline internal/database/metadata/txproxy.go:59]: `return err`

Comment (wfxr) [reply internal/database/metadata/txproxy.go]: fixed

**Comment by YikSanChan** [comment]
@wfxr close this?

**Comment by wfxr** [comment]
Closed by #495

**Pull request closed without merging**